### PR TITLE
Fix bug in bitmap offset computation

### PR DESCRIPTION
**Pull request by lava**
### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: libvast/vast/bitmap_algorithms.hpp [hunk 1]
```diff
@@ -411,13 +411,12 @@ class bitwise_range : public detail::range_facade<bitwise_range<BitRange>> {
          !rng_.done();
          n_ += bits().size(), rng_.next()) {
       VAST_ASSERT(k > 0);
-      if (k > bits().size()) {
-        k -= rank<Bit>(bits());
-      } else {
+      if (k <= bits().size()) {
```
Comment (tobim) [inline libvast/vast/bitmap_algorithms.hpp:414]: I would like to understand why we use `bits().size()` here but decrement by `rank<Bit>(bits())` for the next round. In what circumstance are these 2 values different?

**Review by tobim — COMMENTED** [review]
I think the fix is correct, but I'm not sure whether there is another problem hiding in that function. Question below.

File: libvast/vast/bitmap_algorithms.hpp (hunk 1 above)
Comment (lava) [reply libvast/vast/bitmap_algorithms.hpp:414]: As far as I understand, `rank<Bit>(bits())` counts the number of 1-bits, ie. how many valid entries we're skipping, while `bits().size()` is the distance between the first and last id in the bitmap. So e.g. a bitmap with entries bits 1 and 1000 set to true would have rank 2 and size 1000.

Comment (mavam) [reply libvast/vast/bitmap_algorithms.hpp:414]: I looked at the unit tests and verified that they are written correctly.

To be precise, `rank<Bit>` counts the number of `Bit`-bits where `Bit` is either 0 or 1.

Comment (tobim) [reply libvast/vast/bitmap_algorithms.hpp:414]: I checked if one could use `rank<Bit>(bits())` in the loop condition and that works as well. I can't say which approach has better performance characteristics though, so I'd say let's keep it as it is.

**Review by tobim — APPROVED** [review]
Awesome work getting to the bottom of this!

**Pull request merged**